### PR TITLE
Add properties to Page and PageModel

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs [hunk 1]
```diff
@@ -23,6 +24,31 @@ public abstract class Page : RazorPageBase, IRazorPage
         /// </summary>
         public PageContext PageContext { get; set; }
```
Comment (hishamco) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs]: Shall we add `PageContext` attribute here?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs]: We explicitly initialize this - so we don't need that attribute.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs [hunk 2]
```diff
@@ -37,11 +44,66 @@ public PageArgumentBinder Binder
...
+
+        /// <summary>
+        /// Gets the <see cref="RazorPages.Page"/> instance this model belongs to.
+        /// </summary>
+        public Page Page => PageContext?.Page;
+
+        /// <summary>
+        /// Gets the <see cref="RazorPages.PageContext"/>.
+        /// </summary>
         [PageContext]
```
Comment (hishamco) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: @pranavkm shall we remove it from here?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: The base type for models is optional, so if you wanted to create a custom base type, there needs to be a way to inject the context in to it. `PageModel` uses the same mechanism. The page factory on the other hand assumes you derive from the `Page` type and sets it. https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/DefaultPageFactory.cs#L62-L63

Comment (hishamco) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: I surprised why it's in one place only 😄

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs [hunk 3]
```diff
@@ -30,10 +31,9 @@ public RazorPage()
         }
 
         /// <summary>
-        /// Gets the <see cref="ITempDataDictionary"/> from the <see cref="ViewContext"/>.
+        /// An <see cref="HttpContext"/> representing the current request execution.
         /// </summary>
-        /// <remarks>Returns null if <see cref="ViewContext"/> is null.</remarks>
-        public ITempDataDictionary TempData => ViewContext?.TempData;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs]: Did you delete `TempData`?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs]: It got moved to the base: https://github.com/aspnet/Mvc/pull/6034/files#diff-a4f1d91073db434f385e8192d601f7cdR106

**Pull request closed without merging**